e2e-tests/docs/CI.md

```diff
@@ -0,0 +1,135 @@
+
+- **IBM Cloud Tests:** All nightly tests for the `main`, `1.3`, and `1.2` branches run against the `rhdh-pr-os` OpenShift Container Platform (OCP) cluster on IBM Cloud.
+
+### Additional Nightly Jobs for Main Branch
+
+The nightly job for the `main` branch also runs against:
+
+- **`rhdh-os-1`** (currently OCP 4.14).
+- **`rhdh-os-2`** (currently OCP 4.15).
+
+> **Note:** The output of the nightly runs, including test results and any relevant notifications, is posted on the Slack channel **`#rhdh-e2e-test-alerts`**.
+
+### CI Job Definitions
+
+#### Nightly Test Job
+
+- **Purpose:** Ensure ongoing stability and detect regressions in different environments.
+- **Trigger:** Scheduled to run every night.
+- **Environments:**
+  - **AKS Nightly Tests:** Runs on the `bsCluster`.
+  - **IBM Cloud Nightly Tests:** Runs on the `rhdh-pr-os`, `rhdh-os-1`, and `rhdh-os-2` clusters.
+- **Configurations:**
+  - Tests are executed on both **RBAC** and **non-RBAC** namespaces.
+- **Steps:**
+  1. **Triggering:** Nightly job is triggered on schedule.
+  2. **Environment Setup:** Uses the `openshift-ci-tests.sh` script for setting up the environment.
+     - **Cluster Selection:** Chooses the appropriate cluster based on the job name.
+     - **Resource Configuration:** Sets up namespaces and configures resources.
+     - **Deployment:** Deploys the Red Hat Developer Hub (RHDH) instance and necessary services.
+  3. **Test Execution:**
+     - Runs full test suites using the `yarn` commands.
+     - Tests are executed similarly to the PR tests but may include additional suites.
+     - **Retry Logic:** Individual tests are retried up to 2 times as specified in the Playwright configuration.
+  4. **Artifact Collection:**
+     - Collects and aggregates results.
+     - Stores artifacts for later review for a retention period of **6 months**.
+  5. **Reporting:**
+     - Posts outputs to Slack channel `#rhdh-e2e-test-alerts`.
+     - Generates report.
+- **Artifacts:** Comprehensive test reports, logs, screenshots.
+- **Notifications:** Results posted on Slack.
+
+### Nightly Test Diagram
+
+![Nightly Testing Diagram](./resources/nightly_diagram.svg)
+
+## Configuration and Installation of Testing Environments
+
+### Automation Processes
+
+- **Scripts Used:**
+
+  - **`openshift-ci-tests.sh`**: Automates the setup of the test environment, deployment of RHDH instances, and execution of tests.
+
+- **Automated Steps:**
+  - **Environment Setup:**
+    - **PR Tests:** Ephemeral environments are automatically created and destroyed per test run to ensure a clean state for each test.
+    - **Nightly Tests:** Use long-running clusters with automated updates to reflect production-like environments.
+  - **Deployment:**
+    - RHDH instances are deployed using automated scripts and Helm charts to ensure consistency and repeatability.
```

Comment:
@nilgaar also add what app-configs and dynamic plugins config are used where can user find it.

Reply:
added under _High-Level Overview of `openshift-ci-tests.sh`_. Thank you!